test/e2e/specs/editor/various/autosave.spec.js

```diff
@@ -0,0 +1,351 @@
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		await pageUtils.pressKeys( 'primary+s' );
+		await notice.waitFor();
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBe( 0 );
+	} );
+
+	test( "shouldn't clear local autosave if save fails", async ( {
+		page,
+		pageUtils,
+	} ) => {
+		const notice = page
+			.getByRole( 'button', { name: 'Dismiss this notice' } )
+			.filter( { hasText: 'Draft saved' } );
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'before save' );
+		await pageUtils.pressKeys( 'primary+s' );
+		await notice.waitFor();
+		await page.keyboard.type( ' after save' );
+		await notice.click();
+
+		// Trigger local autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave( { local: true } )
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		await page.route(
+			( url ) =>
+				url.href.includes(
+					`rest_route=${ encodeURIComponent( '/wp/v2/posts/' ) }`
+				),
+			async ( route ) => {
+				await route.abort( 'internetdisconnected' );
+			}
+		);
+		await pageUtils.pressKeys( 'primary+s' );
+
+		await expect(
+			page.locator( '.components-notice__content' )
+		).toContainText( 'Updating failed. You are probably offline.' );
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBe( 1 );
+	} );
+
+	// See https://github.com/WordPress/gutenberg/pull/17501.
+	test( "shouldn't conflict with server-side autosave", async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'before save' );
+		await editor.publishPost();
+
+		const paragraph = editor.canvas.getByRole( 'document', {
+			name: 'Block: Paragraph',
+		} );
+		await paragraph.click();
+		await page.keyboard.type( ' after save' );
+
+		// Trigger remote autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave()
+		);
+		// Force conflicting local autosave.
+		await page.evaluate( () =>
+			window.wp.data.dispatch( 'core/editor' ).autosave( { local: true } )
+		);
+
+		expect(
+			await page.evaluate( () => window.sessionStorage.length )
+		).toBeGreaterThanOrEqual( 1 );
+
+		await page.reload();
+		await page.waitForFunction( () => window?.wp?.data );
+
+		// FIXME: Occasionally, upon reload, there is no server-provided
+		// autosave value available, despite our having previously explicitly
+		// autosaved. The reasons for this are still unknown. Since this is
+		// unrelated to *local* autosave, until we can understand them, we'll
+		// drop this test's expectations if we don't have an autosave object
+		// available.
+		const stillHasRemoteAutosave = await page.evaluate(
+			() =>
+				window.wp.data.select( 'core/editor' ).getEditorSettings()
+					.autosave
```

Comment:
Maybe we could try evaluating with `waitForFunction` here - is it possible that the autosave value is unavailable initially? If yes, a subsequent call via `waitForFunction` could return the correct value.

Reply:
...or via `expect.poll`:

```js
await expect.poll(page.evaluate( () =>
  !! window.wp.data.select( 'core/editor' ).getEditorSettings().autosave
)).toBe( true );
```

Reply:
The value is coming from server-side editor settings, so it should be available initially 🤷‍♂️

Reply:
I have a few new theories and will try to test them tomorrow. While I want to resolve this case, I don't think it's a blocker for migration itself.

Reply:
I created a separate issue, #58959. Let's try fixing this in a follow-up.